lib/src/sign_tool.dart

```diff
@@ -15,44 +15,138 @@ class SignTool {
   final Logger _logger = GetIt.I<Logger>();
   final Configuration _config = GetIt.I<Configuration>();
 
-  /// Use Powershell script to get the Publisher ("Subject") of the certificate
+  /// get the certificate "Subject" for the Publisher value
   Future<void> getCertificatePublisher() async {
     _logger.trace('getting certificate publisher');
 
+    var subject = '';
+
+    if (_config.signToolOptions != null) {
+      if (_config.signToolOptions!.containsArgument('/sha1')) {
+        subject = await _getCertificateSubjectByThumbprint();
+      } else if (_config.signToolOptions!.containsArguments(['/n', '/r'])) {
+        subject = await _getCertificateSubjectBySubject();
+      } else if (_config.signToolOptions!.containsArgument('/i')) {
+        subject = await _getCertificateSubjectByIssuer();
+      }
+    } else if (_config.certificatePath != null &&
+        extension(_config.certificatePath!).toLowerCase() == '.pfx') {
+      subject = await _getPfxCertificateSubject();
+    }
+
+    if (subject.isNotEmpty) {
+      _config.publisher = subject;
+    }
+
+    if (_config.publisher != null && _config.publisher!.isNotEmpty) {
+      _checkCertificateSubject(_config.publisher!);
+    } else {
+      _failToGetCertificateSubject();
+    }
+  }
+
+  /// prints useful information on the Publisher value messing
+  /// and exit the program
+  void _failToGetCertificateSubject() {
+    _logger.stdout('could not find the Publisher value.'.red);
+    _logger.stdout(
+        'you must provide the Publisher value at "msix_config: publisher" in the pubspec.yaml file'
+            .red);
+    _logger.stdout(
+        'the Publisher is the certificate "Subject" in this exact format: "CN=Contoso Software, O=Contoso Corporation, C=US"');
+    _logger.stdout('see where you can found your certificate Subject:');
+    _logger.stdout(
+        'https://drive.google.com/file/d/1oAsnrp2Kf-jZ_kaRjyF5llQ0YZy1IwNe/view?usp=sharing'
```

Comment:
I suggest you store the image into this GitHub repository and link to it via a commit url - to make it permanent.

Reply:
done here https://github.com/YehudaKremer/msix/pull/156/commits/114944835b5b57dcecac0a87541941f7c1650632